### PR TITLE
fix localhost address check in xfcc authenticator

### PR DESCRIPTION
**Pull request by ramaraochavali**
DO NOT CHERRYPICK WITHOUT https://github.com/istio/istio/pull/41171

### CODE REVIEW

**Comment by ramaraochavali** [comment]
@howardjohn PTAL

File: security/pkg/server/ca/authenticate/xfcc_authenticator.go [hunk 1]
```diff
@@ -98,7 +99,12 @@ func isTrustedAddress(addr string, trustedCidrs []string) bool {
 		}
 	}
 	// Always trust local host addresses.
-	return net.ParseIP(addr).IsLoopback()
+	ip, _, err := net.SplitHostPort(addr)
```
Comment (hzxuzhonghu) [inline security/pkg/server/ca/authenticate/xfcc_authenticator.go:102]: why donot split in front, so isInRange can use it

Comment (ramaraochavali) [reply security/pkg/server/ca/authenticate/xfcc_authenticator.go:96]: done

File: security/pkg/server/ca/authenticate/xfcc_authenticator_test.go [hunk 2]
```diff
@@ -130,7 +138,8 @@ func TestXfccAuthenticator(t *testing.T) {
 			if len(tt.xfccHeader) > 0 {
 				md.Append(xfccparser.ForwardedClientCertHeader, tt.xfccHeader)
 			}
-			ctx := peer.NewContext(context.Background(), &peer.Peer{Addr: &net.IPAddr{IP: net.ParseIP("127.0.0.1").To4()}})
+			addr := net.TCPAddrFromAddrPort(netip.MustParseAddrPort("127.0.0.1:2301"))
```
Comment (howardjohn) [inline security/pkg/server/ca/authenticate/xfcc_authenticator_test.go:141]:
```suggestion
			addr := net.TCPAddrFromAddrPort(netip.MustParseAddrPort(tt.peer))
```

unless I am missing something

**Review by howardjohn — APPROVED** [review]

File: security/pkg/server/ca/authenticate/xfcc_authenticator_test.go (hunk 2 above)
Comment (ramaraochavali) [reply security/pkg/server/ca/authenticate/xfcc_authenticator_test.go:141]: There is tt.peer here. Earlier we had this net.ParseIP("127.0.0.1") and now it is changed to net.TCPAddrFromAddrPort(netip.MustParseAddrPort("127.0.0.1:2301"))

**Comment by tlwr** [comment]
Thanks for fixing this :)

File: security/pkg/server/ca/authenticate/xfcc_authenticator_test.go (hunk 2 above)
Comment (ramaraochavali) [reply security/pkg/server/ca/authenticate/xfcc_authenticator_test.go:141]: yes it is hard coded. Let us merge this and I will improve test cases to move it out to test params? If you are OK with that I will remove the `do-not-merge'  label.

**Comment by ramaraochavali** [comment]
@howardjohn I am removing the DNM label. I will follow-up with test case improvements if I find any thing useful

**Pull request merged**